azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusMetadata.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor.container.watch;
+
+import static azkaban.executor.container.KubernetesContainerizedImpl.CLUSTER_LABEL_NAME;
+import static azkaban.executor.container.KubernetesContainerizedImpl.EXECUTION_ID_LABEL_NAME;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.util.Watch;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is intended for maintaining any relevant data from a Pod Watch event that was derived
+ * during the inference of {@link AzPodStatus}. The data here can be consumed by any downstream
+ * callbacks.
+ */
+public class AzPodStatusMetadata {
+  private static final Logger logger = LoggerFactory.getLogger(AzPodStatusMetadata.class);
+
+  private final AzPodStatus azPodStatus;
+  private final String podName;
+  private final Watch.Response<V1Pod> podWatchEvent;
```

Comment:
We should revisit the java process memory for Web server when we are planning to ramp up containerization completely to make sure we are not bringing web server down.

Reply:
Good point. The additional memory usage has been considered as part of the this imlementation - as the individual Watch<V1Pod> events can be a couple of kbs in size. However, as part of this PR we will not be holding on to references to these events and even with few 100s (more than expected) of events 'in-flight' the additional usage should not exceed a couple of MBs. 
A subsequent PR will introduce logic for capturing at least parts of the events for every flow in progress, that will require a more careful analysis of the additional memory usage.